src/lambda_notification.py

```diff
@@ -78,21 +78,27 @@ def lambda_handler(event, context):
 
         message = Message(formatted_message, message_subject)
 
-        secret_client = SecretManager()
-        smtp_secret_name = delivery_options_info.get("smtp_secret_name")
-        smtp_secret = secret_client.get_secret(smtp_secret_name)
-
-        sender = senders.get(
-            delivery_method=delivery_method_type,
-            message=message,
-            ses_sender=delivery_method.get("sender_email"),
-            recipients=delivery_options_info.get("recipients"),
-            smtp_sender=smtp_secret["SMTP_SENDER"],
-            smtp_server=smtp_secret["SMTP_SERVER"],
-            smtp_port=smtp_secret["SMTP_PORT"],
-            smtp_login=smtp_secret["SMTP_LOGIN"],
-            smtp_password=smtp_secret["SMTP_PASSWORD"],
-        )
+        if delivery_method_type == "SMTP":
```

Comment:
on a second thought, can we consider here using SenderProvider?

I mean instead of multiple if-else for each type.
And, actually, it wouldn't be necessary to get secret's data here - it'd be a responsibility of sender (located by sender provider).

what do you think?

Reply:
Yeah, it's possible, just need to pass smtp secret as kwargs, I believe.
I was just not too confident about delegating secrets manager manipulation logic to Sender classes, seemed a bit unnatural for them to take additional responsibility, i.e. retrieving some creds from a separate AWS Service. But can do that to avoid these if-elses which I don't really like either.

Reply:
Maybe we can consider delegating this to SenderProvider, not specific senders.